newsroom/auth/views.py

```diff
@@ -283,3 +308,60 @@ def impersonate_stop():
     start_user_session(user)
     flask.session.pop("auth_user")
     return flask.redirect(flask.url_for("settings.app", app_id="users"))
+
+
+@blueprint.route("/change_password", methods=["GET", "POST"])
+@login_required
+def change_password():
+    form = ResetPasswordForm()
+    user = get_user_required()
+    company = get_company(user)
+    auth_provider = get_company_auth_provider(company)
+    form.email.process_data(user["email"])
+
+    if form.validate_on_submit():
+        if auth_provider.type == AuthProviderType.FIREBASE:
+            if form.data.get("firebase_status"):
+                firebase_status = form.data["firebase_status"]
+                if firebase_status == "OK":
+                    flask.flash(gettext("Your password has been changed."), "success")
+                elif firebase_status == "auth/wrong-password":
+                    flask.flash(gettext("Wrong current password."), "error")
+                else:
+                    logger.warning("Unhandled firebase error %s", firebase_status)
+                    flask.flash(gettext("Could not change password. Please try again later."), "warning")
```

Comment:
I think `Please try again later` might not be good wording here. The user would be thinking if it doesn't work now, how could it work later.
What if we instead suggest to them to contact their administrator?